invoice.py

```diff
@@ -51,27 +51,48 @@ def __register__(cls, module_name):
         Identifier = pool.get('party.identifier')
         identifier = Identifier.__table__()
 
+        exist_tax_identifier = table.column_exist('tax_identifier')
         exist_party = table.column_exist('party')
+
+        if exist_tax_identifier:
+            table.drop_column('tax_identifier')
+
         super(Record, cls).__register__(module_name)
-        if exist_party:
+
+        if exist_tax_identifier or exist_party:
             # Don't use UPDATE FROM because SQLite nor MySQL support it.
-            value = invoice.join(identifier,
-                condition=identifier.id == invoice.tax_identifier).select(
-                    identifier.code,
-                    where=(invoice.id == sql_table.invoice) &
-                    (identifier.type == 'eu_vat'))
+            value = identifier.join(invoice,
+                condition=invoice.party_tax_identifier == identifier.id).select(
+                    identifier.id,
+                    where=(identifier.type == 'eu_vat') &
+                    (invoice.id == sql_table.invoice))
             cursor.execute(*sql_table.update([sql_table.tax_identifier],
                     [value])),
 
+        if exist_tax_identifier:
+            # Update empty tax_identifier with party tax identifier
+            value = identifier.join(party,
```

Comment:
Why do we have two queries for the same purpose?

Reply:
We set the new party_tax_identifier field from different models to ensure that all registers will be filled.

First try from the identifier field of invoice, after from the identifier field of the invoice party and after from the party identifier field.